From 2d2a92e61182d3027932ed48e46097e1cf5ee86c Mon Sep 17 00:00:00 2001
From: Chirag Jain <quic_chirjain@quicinc.com>
Date: Thu, 23 Dec 2021 10:04:49 +0000
Subject: [PATCH] cross-compilation support for gpsd

add cross-compilation support for gpsd by modification of validation of sizeof time_t variable
---
 SConstruct | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/SConstruct b/SConstruct
index 33e0ff326..39e8f44d0 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1043,11 +1043,20 @@ else:
             confdefs.append("/* #undef HAVE_%s_H */\n"
                             % hdr.replace("/", "_").upper())

-    sizeof_time_t = config.CheckSizeOf("time_t")
+    if not env['target']:
+        sizeof_time_t = config.CheckSizeOf("time_t")
+        if 0 < int(sizeof_time_t):
+            announce("sizeof(time_t) is %s" % sizeof_time_t)
+            if 4 >= int(sizeof_time_t):
+                announce("WARNING: time_t is too small.  It will fail in 2038")
+        else:
+            announce("WARNING: could not get sizeof(time_t)")
+            sizeof_time_t = 8
+    else:
+        announce("Not checking sizeof(time_t) when cross-compiling")
+        sizeof_time_t = 8
+
     confdefs.append("#define SIZEOF_TIME_T %s\n" % sizeof_time_t)
-    announce("sizeof(time_t) is %s" % sizeof_time_t)
-    if 4 >= int(sizeof_time_t):
-        announce("WARNING: time_t is too small.  It will fail in 2038")

     # check function after libraries, because some function require libraries
     # for example clock_gettime() require librt on Linux glibc < 2.17
--
2.25.1
